### ISSUE TITLE
[Bug]: showSaveDialog never resolves when dialog is cancelled by clicking outside (macOS)

### ISSUE THREAD

**Issue by ybouane** [electron/electron#38796]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

25.1.0

### What operating system are you using?

macOS

### Operating System Version

macOS Ventura 13.3 (22E252)

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

When clicking outside of the dialog, the default behaviour on macOS is to cancel the dialog. Electron should resolve the call with {canceled: true} just like it does when explicitly clicking on the cancel button.

### Actual Behavior

When clicking on the cancel button, it resolves with {canceled: true}, but not when clicking outside the dialog. Instead it just never resolves (neither does it throw an error).

**Comment by codebytere** [comment]
it's not clear here what you mean by:

> When clicking outside of the dialog, the default behaviour on macOS is to cancel the dialog.

In which native apps are you seeing this behavior? You need to interact with the dialog to dismiss it - i'm not aware of any native apps where simply clicking outside the dialog closes it.

**Comment by ybouane** [comment]
That is very strange, it seems to be working now..

The issue was that when I would click on the browserWindow (blue area here):
![image](https://github.com/electron/electron/assets/14046002/4bceb730-0f90-48be-9108-17d9555525c9)

The dialog would close (I actually checked behind the dialog to see if it was hidden behind the browser window, I could swear that it wasn't). It was closing without resolving the promise. I even tried a few times to double check and when clicking the browserWindow, it was nowhere to be found.

Now when I tried again, it just went behind the browser window!

I have now fixed the issue by referencing the window to which I need the dialog to be attached to. That way, it's always on top.

Sorry for this issue.
